unified-runtime/source/loader/layers/sanitizer/sanitizer_common/sanitizer_utils.hpp

```diff
@@ -59,11 +61,16 @@ size_t GetKernelPrivateMemorySize(ur_kernel_handle_t Kernel,
 size_t GetVirtualMemGranularity(ur_context_handle_t Context,
                                 ur_device_handle_t Device);
 
+template <class T>
 ur_result_t
-EnqueueUSMBlockingSet(ur_queue_handle_t Queue, void *Ptr, char Value,
-                      size_t Size, uint32_t NumEvents = 0,
+EnqueueUSMBlockingSet(ur_queue_handle_t Queue, void *Ptr, T Value, size_t Size,
+                      uint32_t NumEvents = 0,
                       const ur_event_handle_t *EventWaitList = nullptr,
-                      ur_event_handle_t *OutEvent = nullptr);
+                      ur_event_handle_t *OutEvent = nullptr) {
+  assert(Size % sizeof(T) == 0);
+  return getContext()->urDdiTable.Enqueue.pfnUSMFill(
```

Comment:
This function doesn't block so the helper name is kind of misleading. I'd suggest adding a urQueueFinish to ensure the operation completes before returning, or renaming the helper so it's understood that the fill won't necessarily be complete after calling it.